### ISSUE TITLE
fatal error: logic_export.h: No such file or directory

### ISSUE THREAD

**Issue by unlawfullyawful** [AfoninZ/MultiMC5-Cracked#61]
### Operating System

Linux

### Description of bug

after 27% compilation this error appears and the compilation is terminated 
![image](https://user-images.githubusercontent.com/81036269/127809951-e5f90633-d12f-4520-9c2d-90de127ba6d9.png)
![image](https://user-images.githubusercontent.com/81036269/127809982-3547d457-3cd9-4dda-9420-38be7871b13f.png)

### Steps to reproduce

this happens during the last step of the building from command line build.md script, make -j8 install where 8 is the number of threads

### This issue is unique

- [X] I have searched the issue tracker and did not find an issue describing my bug.

**Comment by Sebastian-byte** [comment]
Yes, we already know this issue, mainly due to "MultiMC" debrand.

**Comment by AfoninZ** [comment]
@Sebastian-byte btw, is it possible to keep as much of the original names internally while only stripping the MultiMC branding from the UI/translations and the crash logs?

**Comment by Sebastian-byte** [comment]
> @Sebastian-byte btw, is it possible to keep as much of the original names internally while only stripping the MultiMC branding from the UI/translations and the crash logs?

I think we can keep a few internal references.

**Comment by Sebastian-byte** [comment]
This problem has been fixed but we still need to fix the providers system.

**Comment by AndreVuillemot160** [comment]
Okie, nice

**Comment by Samisaprogrammer** [comment]
Hello, now the build is failing because login radio buttons are not defined in their scope.

**Comment by Sebastian-byte** [comment]
> Hello, now the build is failing because login radio buttons are not defined in their scope.

I'm trying to fix it currently.

**Comment by Samisaprogrammer** [comment]
This is where it failed, btw https://github.com/AfoninZ/MultiMC5-Cracked/commit/65d3068e9fb0798a7e5383377082ee45bafb3d6b

**Comment by Slddev** [comment]
You guys should only work on the stable branch so you can catch up

**Comment by Sebastian-byte** [comment]
> You guys should only work on the stable branch so you can catch up

That actually makes it more complicated since there are major differences between commits in the branch.

**Comment by Slddev** [comment]
> > You guys should only work on the stable branch so you can catch up
> 
> That actually makes it more complicated since there are major differences between commits in the branch.

Oh ok.

**Comment by Slddev** [comment]
> Ok so I've figured out how to fix half the build.
> 
> Adding `QMap<QString, QRadioButton*> m_radioButtons;` to `launcher/dialogs/LoginDialog.h` line 58 fixes a small bit of the problem.

Nicely done! This is the new error:

```bash
D:\a\MultiMC5-Cracked\MultiMC5-Cracked\launcher\dialogs\LoginDialog.cpp: In member function 'virtual void LoginDialog::accept()':
D:\a\MultiMC5-Cracked\MultiMC5-Cracked\launcher\dialogs\LoginDialog.cpp:56:24: error: 'class MinecraftAccount' has no member named 'setProvider'
             m_account->setProvider(AuthProviders::lookup(providerId));
                        ^
mingw32-make[2]: *** [launcher/CMakeFiles/Launcher_logic.dir/dialogs/LoginDialog.cpp.obj] Error 1
mingw32-make[2]: *** Waiting for unfinished jobs....
launcher\CMakeFiles\Launcher_logic.dir\build.make:3957: recipe for target 'launcher/CMakeFiles/Launcher_logic.dir/dialogs/LoginDialog.cpp.obj' failed
mingw32-make[1]: *** [launcher/CMakeFiles/Launcher_logic.dir/all] Error 2
CMakeFiles\Makefile2:1984: recipe for target 'launcher/CMakeFiles/Launcher_logic.dir/all' failed
mingw32-make: *** [all] Error 2
makefile:144: recipe for target 'all' failed
```

**Comment by Slddev** [comment]
It worked for me in my fork.

**Comment by Samisafool** [comment]
@Sebastian-byte the build probably works on local, this is an issue with mmc itself, to fix it, remove the following:

https://github.com/AfoninZ/MultiMC5-Cracked/blob/f2f9ded41d345bb2296231c9482a1da41542253a/.github/workflows/main.yml#L37

https://github.com/AfoninZ/MultiMC5-Cracked/blob/f2f9ded41d345bb2296231c9482a1da41542253a/.github/workflows/main.yml#L158

https://github.com/AfoninZ/MultiMC5-Cracked/blob/f2f9ded41d345bb2296231c9482a1da41542253a/.github/workflows/main.yml#L246

and it should work

**Comment by Sebastian-byte** [comment]
Yes, I noticed it just now, we can just wait until the upstream re-enables guest authentication in the build server.

**Comment by Samisafool** [comment]
What is that?

**Comment by Sebastian-byte** [comment]
The TeamCity build server used by the upstream, and where we get the build number, it seems like they disabled guest authentication temporally, one solution is to wait until they re-enable it.

**Comment by Samisafool** [comment]
So I built locally, and turns out that it was failing. After fixing that, I'm not sure where to proceed next.

````src/launcher/MainWindow.cpp: In member function ‘void MainWindow::retranslateUi()’:
src/launcher/MainWindow.cpp:849:29: error: ‘profileInUseFilter’ was not declared in this scope
  849 |         auto profileLabel = profileInUseFilter(defaultAccount->profileName(), defaultAccount->isInUse());
      |                             ^~~~~~~~~~~~~~~~~~
[ 29%] Building CXX object launcher/CMakeFiles/Launcher_logic.dir/meta/BaseEntity.cpp.o
src/launcher/MainWindow.cpp: In member function ‘void MainWindow::repopulateAccountsMenu()’:
src/launcher/MainWindow.cpp:1035:33: error: ‘profileInUseFilter’ was not declared in this scope
 1035 |             auto profileLabel = profileInUseFilter(defaultAccount->profileName(), defaultAccount->isInUse());
      |                                 ^~~~~~~~~~~~~~~~~~
````
